flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -534,215 +557,235 @@ private Configuration applyYarnProperties(Configuration configuration) throws Fl
+						yarnApplicationId,
+						new ScheduledExecutorServiceAdapter(scheduledExecutorService));
+
+					try {
 						runInteractiveCli(
-							yarnCluster,
+							clusterClient,
 							yarnApplicationStatusMonitor,
 							acceptInteractiveInput);
-					} catch (Exception e) {
-						LOG.info("Could not properly close the Yarn application status monitor.", e);
 					} finally {
+						try {
+							yarnApplicationStatusMonitor.close();
+						} catch (Exception e) {
+							LOG.info("Could not properly close the Yarn application status monitor.", e);
+						}
+
+						try {
+							clusterClient.shutdown();
+						} catch (Exception e) {
+							LOG.info("Could not properly shutdown cluster client.", e);
+						}
+
+						try {
+							yarnClusterDescriptor.terminateCluster(yarnApplicationId);
+						} catch (FlinkException e) {
+							LOG.info("Could not properly terminate the Flink cluster.", e);
+						}
+
 						// shut down the scheduled executor service
 						ExecutorUtils.gracefulShutdown(
 							1000L,
 							TimeUnit.MILLISECONDS,
 							scheduledExecutorService);
+
+						deleteYarnPropertiesFile();
+
+						try {
+							final ApplicationReport applicationReport = yarnClusterDescriptor
+								.getYarnClient()
+								.getApplicationReport(yarnApplicationId);
+
+							logFinalApplicationReport(applicationReport);
+						} catch (YarnException | IOException e) {
+							LOG.info("Could not log the final application report.", e);
+						}
 					}
 				}
-			} catch (FlinkException e) {
-				System.err.println("Error while deploying a Flink cluster: " + e.getMessage());
-				e.printStackTrace();
-				return 1;
+			}
+		} finally {
+			try {
+				yarnClusterDescriptor.close();
+			} catch (Exception e) {
+				LOG.info("Could not properly close the yarn cluster descriptor.", e);
 			}
 		}
+
 		return 0;
 	}
 
-	/**
-	 * Utility method for tests.
-	 */
-	public void stop() {
-		if (yarnCluster != null) {
-			LOG.info("Command line interface is shutting down the yarnCluster");
+	private void logFinalApplicationReport(ApplicationReport appReport) {
+		try {
+			LOG.info("Application " + appReport.getApplicationId() + " finished with state " + appReport
+				.getYarnApplicationState() + " and final state " + appReport
+				.getFinalApplicationStatus() + " at " + appReport.getFinishTime());
+
+			if (appReport.getYarnApplicationState() == YarnApplicationState.FAILED || appReport.getYarnApplicationState()
+				== YarnApplicationState.KILLED) {
+				LOG.warn("Application failed. Diagnostics " + appReport.getDiagnostics());
+				LOG.warn("If log aggregation is activated in the Hadoop cluster, we recommend to retrieve "
+					+ "the full application log using this command:"
+					+ System.lineSeparator()
+					+ "\tyarn logs -applicationId " + appReport.getApplicationId()
+					+ System.lineSeparator()
+					+ "(It sometimes takes a few seconds until the logs are aggregated)");
+			}
+		} catch (Exception e) {
```

Comment:
You are just logging some information. It should not throw an exception.

Reply:
True. Will fix it.